mm2src/coins/nft.rs

```diff
@@ -624,12 +680,61 @@ async fn get_moralis_nft_list(
             } else {
                 break;
             }
+        } else {
+            break;
         }
     }
     drop_mutability!(res_list);
     Ok(res_list)
 }
 
+pub(crate) async fn get_nfts_for_activation(
+    chain: &Chain,
+    my_address: &Address,
+    url: &Url,
+) -> MmResult<HashMap<String, NftInfo>, GetNftInfoError> {
+    let mut nfts_map = HashMap::new();
+    let uri_without_cursor = construct_moralis_uri_for_nft(url, &eth_addr_to_hex(my_address), chain)?;
+
+    // The cursor returned in the previous response (used for getting the next page).
+    let mut cursor = String::new();
+    loop {
+        let uri = format!("{}{}", uri_without_cursor, cursor);
+        let response = send_request_to_uri(uri.as_str()).await?;
+        if let Some(nfts_list) = response["result"].as_array() {
+            for nft_json in nfts_list {
+                let nft_moralis: NftFromMoralis = serde_json::from_str(&nft_json.to_string())?;
+                let contract_type = match nft_moralis.contract_type {
+                    Some(contract_type) => contract_type,
+                    None => continue,
+                };
+                let token_address_str = eth_addr_to_hex(&nft_moralis.common.token_address);
+                let nft_info = NftInfo {
+                    token_address: nft_moralis.common.token_address,
+                    token_id: nft_moralis.token_id.0.clone(),
+                    chain: *chain,
+                    contract_type,
+                    amount: nft_moralis.common.amount,
+                };
+                let key = format!("{},{}", token_address_str, nft_moralis.token_id.0);
+                nfts_map.insert(key, nft_info);
+            }
+            // if cursor is not null, there are other NFTs on next page,
+            // and we need to send new request with cursor to get info from the next page.
+            if let Some(cursor_res) = response["cursor"].as_str() {
+                cursor = format!("{}{}", "&cursor=", cursor_res);
+                continue;
+            } else {
+                break;
+            }
+        } else {
+            break;
+        }
+    }
+    drop_mutability!(nfts_map);
```

Comment:
not needed since you returning the value immediately 👀

Reply:
lol, yeah I wrote it reflexively. removed all `drop_mutability` which were at the end of functions